unittest/test_core_nn.sh.in

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash -x
+# test_core_nn.sh
+#
+
+# Some tests for core NN, e.g. generation of test data using
+# dump_data, and unquantised synthesis using and test_lpcnet.  Used to
+# ensure no existing features are broken during experimentation and
+# development.
+
+# test generation of training data (doesn't really test training as that takes hours)
+# TODO: This test not working yet
+if [ ! -z $TRAIN_TEST ]; then
+    TRAIN_SRC=all_speech
+    TRAIN_OUT_PCM=$(mktemp).pcm
+    TRAIN_OUT_F32=$(mktemp).f32
+    TRAIN_TARG_PCM=all_speech.pcm
+    TRAIN_TARG_F32=all_speech.f32
+    @CMAKE_CURRENT_BINARY_DIR@/src/dump_data --train --c2pitch -z 0 -n 1E6 ~/Downloads/$TRAIN_SRC.sw $TRAIN_OUT_F32 $TRAIN_OUT_PCM
```

Comment:
I'm not fond of the `'.in` -> `.sh` style as it means the test can't be run manually without running `cmake` after every edit.  There are some more convenient models in the `codec2` tests, e.g. setting the `$PATH` externally or passing in a build time path variable.

Having said that - most of these tests don't work anyway.  I'll take an action to add a simple test that at least tests the changes in this PR.